File: wallet/src/main/java/com/worldventures/wallet/analytics/firmware/action/FirmwareAnalyticsAction.java
package com.worldventures.wallet.analytics.firmware.action;

import android.support.annotation.CallSuper;
import android.support.annotation.NonNull;

import com.worldventures.core.service.analytics.Attribute;
import com.worldventures.core.service.analytics.BaseAnalyticsAction;
import com.worldventures.wallet.domain.entity.FirmwareUpdateData;

public abstract class FirmwareAnalyticsAction extends BaseAnalyticsAction {

   @Attribute("cid") String cid;

   @CallSuper
   public void setFirmwareData(@NonNull FirmwareUpdateData data) {
      cid = data.smartCardId();
   }
}

File: wallet/src/main/java/com/worldventures/wallet/analytics/firmware/action/FirmwareAnalyticsAction.kt
package com.worldventures.wallet.analytics.firmware.action

import android.support.annotation.CallSuper

import com.worldventures.core.service.analytics.Attribute
import com.worldventures.core.service.analytics.BaseAnalyticsAction
import com.worldventures.wallet.domain.entity.FirmwareUpdateData

abstract class FirmwareAnalyticsAction : BaseAnalyticsAction() {

   @Attribute("cid") internal lateinit var cid: String

   @CallSuper
   open fun setFirmwareData(data: FirmwareUpdateData) {
      cid = data.smartCardId()
   }
}
